src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -150,6 +153,33 @@ static int client_receive_message_udp(
                 void *userdata);
 static void client_stop(sd_dhcp_client *client, int error);
 
+int sd_dhcp_client_id_to_string(char **ret, const void *data, size_t len) {
+        const sd_dhcp_client_id *client_id = data;
+
+        switch (client_id->type) {
+        case 0:
+                if (utf8_is_printable((char *)client_id->gen.data, len)) {
+                        return asprintf(ret, "%.*s", (int) len, client_id->gen.data);
```

Comment:
`asprintf` may fail. This needs to use an intermediate variable:
```C
_clenaup_free_ char *t = NULL;
if (asprintf(&t, ...) < 0)
     return -ENOMEM;
...
*ret = TAKE_PTR(t);
return 0;
```

Reply:
done